server/src/main/java/org/elasticsearch/search/rescore/RescorerBuilder.java

```diff
@@ -68,6 +68,24 @@ public Integer windowSize() {
         return windowSize;
     }
 
+    /**
+     * In some situations (e.g., LTR rescorer), it is impossible to combine scores issued by the rescoring phase those from
+     * the first-pass query (or previous rescorers) because they are not comparable with each other.
+     *
+     * In this case:
+     *
+     * - we need to ensure that the full topDocs is rescored
+     * - the topDocs is truncated to the window size before executing the rescorer
+     * - we prevent subsequent rescorers with a bigger window size
+     * - we check the window size for the rescorer is at least equals to from + size
+     * - window size is a required parameter for the rescorer
+     *
+     * @return whether it is possible to combine scores issued by the rescoring phase with original scores or not.
+     */
+    public boolean canCombineScores() {
```

Comment:
My gut reaction is that this is a bad idea.

At least for the validation part, we should instead add a `validate` method to the builders.

As for handling the logic later, more thought needs to be done there.